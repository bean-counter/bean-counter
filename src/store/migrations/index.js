import uuid from 'uuid/v4'

const migrations = {
  // Add accountType to accounts
  0: (state) => {
    if (!state.accounts) return state
    return {
      ...state,
      accounts: {
        ...state.accounts,
        byId: Object.values(state.accounts.byId).reduce((result, account) => ({
          ...result,
          [account.id]: { ...account, accountType: 'Bank' }
        }), {})
      }
    }
  },
  // Cleanup settings (remove snackbarMessage)
  1: (state) => {
    if (!state.settings) return state
    return {
      ...state,
      settings: {
        currency: state.settings.currency,
        locale: state.settings.locale
      }
    }
  },
  // Update transaction amount
  // Remove dates array from exchangeRates
  2: (state) => {
    if (!state.transactions) return state
    return {
      ...state,
      transactions: {
        list: state.transactions.list.map((transaction) => ({
          ...transaction,
          amount: { accountCurrency: transaction.amount, localCurrency: null }
        }))
      },
      exchangeRates: Object.keys(state.exchangeRates).reduce((result, currency) => {
        const { dates, ...exchangeRates } = state.exchangeRates[currency]
        return {
          ...result,
          [currency]: exchangeRates
        }
      }, {})
    }
  }
  // 3: (state) => {
  //   if (!state.budget) return state
  //   // collect accountIds
  //   state.transactions.list.forEach((transaction) => {
  //     if (transaction.categoryId === categoryId) {

  //     }
  //   })

  //   return {
  //     ...state,
  //     budget: {
  //       ...state.budget,
  //       rules: Object.keys(state.budget.rules).reduce((res, match) => {
  //         const ruleId = uuid()
  //         const { categoryId } = state.budget.rules[match]
  //         return {
  //           id: ruleId,
  //           accountId: account.id,
  //           attributes: { categoryId },
  //           filterBy: { description: { type: 'equals', value: match } }
  //         }
  //       })
  //     },
  //     exchangeRates: Object.keys(state.exchangeRates).reduce((result, currency) => {
  //       const { dates, ...exchangeRates } = state.exchangeRates[currency]
  //       return {
  //         ...result,
  //         [currency]: exchangeRates
  //       }
  //     }, {})
  //   }
  // }
}

export default migrations
