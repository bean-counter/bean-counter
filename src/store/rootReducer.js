import { combineReducers } from 'redux'
import userReducer from './user/reducer'
import transactionsReducer from './transactions/reducer'
import marketValuesReducer from './marketValues/reducer'

export default combineReducers({
  user: userReducer,
  transactions: transactionsReducer,
  marketValues: marketValuesReducer
})
