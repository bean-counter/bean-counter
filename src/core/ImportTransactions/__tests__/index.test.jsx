import React from 'react'
import renderer from 'react-test-renderer'
import { BrowserRouter } from 'react-router-dom'
import { Provider } from 'react-redux'
import store from '../../../store'
import { ImportTransactionsComponent } from '../'

jest.mock('../../../common/Header', () => 'Header')

describe('Import Transactions', () => {
  beforeEach(() => {
    jest.clearAllMocks()
  })

  const mockSaveTransactions = jest.fn()

  it('matches snapshot', () => {
    const component = renderer.create((
      <Provider store={store}>
        <BrowserRouter>
          <ImportTransactionsComponent
            classes={{ }}
            saveTransactions={mockSaveTransactions}
          />
        </BrowserRouter>
      </Provider>
    ))
    expect(component.toJSON()).toMatchSnapshot()
  })
})
