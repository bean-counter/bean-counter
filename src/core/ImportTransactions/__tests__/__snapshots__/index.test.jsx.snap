// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Import Transactions matches snapshot 1`] = `
<div>
  <div>
    <div
      className="Header-root-1"
    >
      <header
        className="MuiPaper-root-13 MuiPaper-elevation4-19 MuiAppBar-root-4 MuiAppBar-positionFixed-5 MuiAppBar-colorPrimary-11 mui-fixed Header-header-2"
      >
        <div
          className="MuiToolbar-root-40 MuiToolbar-regular-42 MuiToolbar-gutters-41 Header-toolbar-3"
        >
          <div
            className="Logo-logo-44"
          >
            <img
              alt="Entaxy logo"
              className="Logo-image-45"
              src="logo.png"
            />
            <a
              className="MuiTypography-root-47 MuiTypography-h6-64 MuiTypography-colorInherit-76 Logo-title-46"
              href="/"
              onClick={[Function]}
            >
              Entaxy
            </a>
          </div>
          <div>
            <a
              aria-current={null}
              className="MuiButtonBase-root-108 MuiButton-root-82 MuiButton-text-84 MuiButton-flat-87 MuiButton-sizeSmall-105 MuiButton-colorInherit-103"
              href="/portfolio"
              onBlur={[Function]}
              onClick={[Function]}
              onContextMenu={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              role="button"
              tabIndex="0"
            >
              <span
                className="MuiButton-label-83"
              >
                Portfolio
              </span>
              <span
                className="MuiTouchRipple-root-196"
              />
            </a>
            <a
              aria-current={null}
              className="MuiButtonBase-root-108 MuiButton-root-82 MuiButton-text-84 MuiButton-flat-87 MuiButton-sizeSmall-105 MuiButton-colorInherit-103"
              href="/import-transactions"
              onBlur={[Function]}
              onClick={[Function]}
              onContextMenu={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              role="button"
              tabIndex="0"
            >
              <span
                className="MuiButton-label-83"
              >
                Import Transactions
              </span>
              <span
                className="MuiTouchRipple-root-196"
              />
            </a>
          </div>
          <div
            className="LoginButtonComponent-root-111"
          >
            <button
              aria-owns={null}
              className="MuiButtonBase-root-108 MuiButton-root-82 MuiButton-text-84 MuiButton-flat-87 MuiButton-colorInherit-103"
              disabled={false}
              onBlur={[Function]}
              onClick={[Function]}
              onContextMenu={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex="0"
              type="button"
            >
              <span
                className="MuiButton-label-83"
              />
              <span
                className="MuiTouchRipple-root-196"
              />
            </button>
          </div>
        </div>
      </header>
      <div
        className="MuiDrawer-root-114 MuiDrawer-docked-115"
      >
        <div
          className="MuiPaper-root-13 MuiPaper-elevation0-15 MuiDrawer-paper-116 withRouter-LeftDrawer--drawerPaper-113 MuiDrawer-paperAnchorLeft-117 MuiDrawer-paperAnchorDockedLeft-121"
        >
          <div
            className="withRouter-LeftDrawer--toolbar-112"
          />
          <ul
            className="MuiList-root-126 MuiList-padding-127"
          >
            <a
              aria-current={null}
              className="MuiButtonBase-root-108 MuiListItem-root-130 MuiListItem-default-133 MuiListItem-gutters-138 MuiListItem-button-139"
              href="/dashboard"
              onBlur={[Function]}
              onClick={[Function]}
              onContextMenu={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              role="button"
              tabIndex="0"
            >
              <div
                className="MuiListItemIcon-root-142"
              >
                <svg
                  aria-hidden="true"
                  className="MuiSvgIcon-root-143"
                  focusable="false"
                  role="presentation"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M0 0h24v24H0z"
                    fill="none"
                  />
                  <path
                    d="M3 13h8V3H3v10zm0 8h8v-6H3v6zm10 0h8V11h-8v10zm0-18v6h8V3h-8z"
                  />
                </svg>
              </div>
              <div
                className="MuiListItemText-root-152"
              >
                <span
                  className="MuiTypography-root-47 MuiTypography-subheading-54 MuiListItemText-primary-155"
                >
                  Dashboard
                </span>
              </div>
              <span
                className="MuiTouchRipple-root-196"
              />
            </a>
          </ul>
          <hr
            className="MuiDivider-root-158"
          />
          <nav
            className="MuiList-root-126 MuiList-dense-128 MuiList-padding-127 MuiList-subheader-129"
          >
            <div
              className="MuiListSubheader-root-168 MuiListSubheader-sticky-173 MuiListSubheader-gutters-171"
            >
              Accounts
              <div
                className="MuiListItemSecondaryAction-root-174"
              >
                <a
                  aria-current={null}
                  aria-describedby={null}
                  aria-label="New account"
                  className="MuiButtonBase-root-108 MuiIconButton-root-183 withRouter-AccountsComponent--smallButton-166"
                  href="/accounts/new"
                  onBlur={[Function]}
                  onClick={[Function]}
                  onContextMenu={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onKeyUp={[Function]}
                  onMouseDown={[Function]}
                  onMouseLeave={[Function]}
                  onMouseOver={[Function]}
                  onMouseUp={[Function]}
                  onTouchEnd={[Function]}
                  onTouchMove={[Function]}
                  onTouchStart={[Function]}
                  role="button"
                  tabIndex="0"
                  title="New account"
                >
                  <span
                    className="MuiIconButton-label-188"
                  >
                    <svg
                      aria-hidden="true"
                      className="MuiSvgIcon-root-143"
                      focusable="false"
                      role="presentation"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
                      />
                      <path
                        d="M0 0h24v24H0z"
                        fill="none"
                      />
                    </svg>
                  </span>
                  <span
                    className="MuiTouchRipple-root-196"
                  />
                </a>
              </div>
            </div>
            <ul
              className="MuiList-root-126 MuiList-dense-128 MuiList-padding-127"
            />
          </nav>
        </div>
      </div>
    </div>
    <h5
      className="MuiTypography-root-47 MuiTypography-h5-63 MuiTypography-alignCenter-70"
    >
      Import transactions
    </h5>
    <div>
      <p
        className="MuiTypography-root-47 MuiTypography-body1-56 MuiTypography-alignCenter-70"
      >
        Select an institution
      </p>
      <div>
        <div>
          <div
            className="MuiPaper-root-13 MuiPaper-elevation1-16 MuiPaper-rounded-14 MuiCard-root-189"
          >
            <div
              className="MuiCardHeader-root-190 undefined undefined"
            >
              <div
                className="MuiCardHeader-content-193"
              >
                <span
                  className="MuiTypography-root-47 MuiTypography-headline-52 MuiCardHeader-title-194"
                >
                  RBC
                </span>
                <span
                  className="MuiTypography-root-47 MuiTypography-body1-56 MuiTypography-colorTextSecondary-80 MuiCardHeader-subheader-195"
                >
                  Royal Bank
                </span>
              </div>
              <div
                className="MuiCardHeader-action-192"
              >
                <button
                  className="MuiButtonBase-root-108 MuiIconButton-root-183"
                  disabled={false}
                  onBlur={[Function]}
                  onClick={[Function]}
                  onContextMenu={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onKeyUp={[Function]}
                  onMouseDown={[Function]}
                  onMouseLeave={[Function]}
                  onMouseUp={[Function]}
                  onTouchEnd={[Function]}
                  onTouchMove={[Function]}
                  onTouchStart={[Function]}
                  tabIndex="0"
                  type="button"
                >
                  <span
                    className="MuiIconButton-label-188"
                  >
                    <svg
                      aria-hidden="true"
                      className="MuiSvgIcon-root-143"
                      focusable="false"
                      role="presentation"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M0 0h24v24H0z"
                        fill="none"
                      />
                      <path
                        d="M12 8c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2zm0 2c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm0 6c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2z"
                      />
                    </svg>
                  </span>
                  <span
                    className="MuiTouchRipple-root-196"
                  />
                </button>
              </div>
            </div>
          </div>
        </div>
        <div>
          <div
            className="MuiPaper-root-13 MuiPaper-elevation1-16 MuiPaper-rounded-14 MuiCard-root-189"
          >
            <div
              className="MuiCardHeader-root-190 undefined undefined"
            >
              <div
                className="MuiCardHeader-content-193"
              >
                <span
                  className="MuiTypography-root-47 MuiTypography-headline-52 MuiCardHeader-title-194"
                >
                  BMO
                </span>
                <span
                  className="MuiTypography-root-47 MuiTypography-body1-56 MuiTypography-colorTextSecondary-80 MuiCardHeader-subheader-195"
                >
                  Bank of Montreal
                </span>
              </div>
              <div
                className="MuiCardHeader-action-192"
              >
                <button
                  className="MuiButtonBase-root-108 MuiIconButton-root-183"
                  disabled={false}
                  onBlur={[Function]}
                  onClick={[Function]}
                  onContextMenu={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onKeyUp={[Function]}
                  onMouseDown={[Function]}
                  onMouseLeave={[Function]}
                  onMouseUp={[Function]}
                  onTouchEnd={[Function]}
                  onTouchMove={[Function]}
                  onTouchStart={[Function]}
                  tabIndex="0"
                  type="button"
                >
                  <span
                    className="MuiIconButton-label-188"
                  >
                    <svg
                      aria-hidden="true"
                      className="MuiSvgIcon-root-143"
                      focusable="false"
                      role="presentation"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M0 0h24v24H0z"
                        fill="none"
                      />
                      <path
                        d="M12 8c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2zm0 2c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm0 6c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2z"
                      />
                    </svg>
                  </span>
                  <span
                    className="MuiTouchRipple-root-196"
                  />
                </button>
              </div>
            </div>
          </div>
        </div>
        <div>
          <div
            className="MuiPaper-root-13 MuiPaper-elevation1-16 MuiPaper-rounded-14 MuiCard-root-189"
          >
            <div
              className="MuiCardHeader-root-190 undefined undefined"
            >
              <div
                className="MuiCardHeader-content-193"
              >
                <span
                  className="MuiTypography-root-47 MuiTypography-headline-52 MuiCardHeader-title-194"
                >
                  TD
                </span>
                <span
                  className="MuiTypography-root-47 MuiTypography-body1-56 MuiTypography-colorTextSecondary-80 MuiCardHeader-subheader-195"
                >
                  TD Canada Trust
                </span>
              </div>
              <div
                className="MuiCardHeader-action-192"
              >
                <button
                  className="MuiButtonBase-root-108 MuiIconButton-root-183"
                  disabled={false}
                  onBlur={[Function]}
                  onClick={[Function]}
                  onContextMenu={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onKeyUp={[Function]}
                  onMouseDown={[Function]}
                  onMouseLeave={[Function]}
                  onMouseUp={[Function]}
                  onTouchEnd={[Function]}
                  onTouchMove={[Function]}
                  onTouchStart={[Function]}
                  tabIndex="0"
                  type="button"
                >
                  <span
                    className="MuiIconButton-label-188"
                  >
                    <svg
                      aria-hidden="true"
                      className="MuiSvgIcon-root-143"
                      focusable="false"
                      role="presentation"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M0 0h24v24H0z"
                        fill="none"
                      />
                      <path
                        d="M12 8c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2zm0 2c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm0 6c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2z"
                      />
                    </svg>
                  </span>
                  <span
                    className="MuiTouchRipple-root-196"
                  />
                </button>
              </div>
            </div>
          </div>
        </div>
        <div>
          <div
            className="MuiPaper-root-13 MuiPaper-elevation1-16 MuiPaper-rounded-14 MuiCard-root-189"
          >
            <div
              className="MuiCardHeader-root-190 undefined undefined"
            >
              <div
                className="MuiCardHeader-content-193"
              >
                <span
                  className="MuiTypography-root-47 MuiTypography-headline-52 MuiCardHeader-title-194"
                >
                  Tangerine
                </span>
                <span
                  className="MuiTypography-root-47 MuiTypography-body1-56 MuiTypography-colorTextSecondary-80 MuiCardHeader-subheader-195"
                >
                  Personal Banking
                </span>
              </div>
              <div
                className="MuiCardHeader-action-192"
              >
                <button
                  className="MuiButtonBase-root-108 MuiIconButton-root-183"
                  disabled={false}
                  onBlur={[Function]}
                  onClick={[Function]}
                  onContextMenu={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onKeyUp={[Function]}
                  onMouseDown={[Function]}
                  onMouseLeave={[Function]}
                  onMouseUp={[Function]}
                  onTouchEnd={[Function]}
                  onTouchMove={[Function]}
                  onTouchStart={[Function]}
                  tabIndex="0"
                  type="button"
                >
                  <span
                    className="MuiIconButton-label-188"
                  >
                    <svg
                      aria-hidden="true"
                      className="MuiSvgIcon-root-143"
                      focusable="false"
                      role="presentation"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M0 0h24v24H0z"
                        fill="none"
                      />
                      <path
                        d="M12 8c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2zm0 2c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm0 6c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2z"
                      />
                    </svg>
                  </span>
                  <span
                    className="MuiTouchRipple-root-196"
                  />
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
